app/controllers/supplementary/SummaryPageController.scala

```diff
@@ -118,4 +118,19 @@ class SummaryPageController @Inject()(
   private def prepareFlashScope(lrn: String) =
     Flash(Map("LRN" -> lrn))
 
+  //TODO : refactor to handle large data collection and move the logic of metadata creation to separate service
+  private def createMetaData(cacheMap: CacheMap, suppDecData: SupplementaryDeclarationData): MetaData = {
```

Comment:
This method is not tested anywhere despite being crucial for the whole journey.
Moving it to a separate class/object is a good idea and it will be easier to test.

Reply:
agree

Reply:
I will have to refactor this file based on the changes that happen with CEDS-1009 I started working on it and will raise a PR